src/Adapter/Image/ImageGenerator.php

```diff
@@ -96,12 +115,29 @@ protected function resize(string $filePath, ImageType $imageType): bool
             $height *= 2;
         }
 
-        return ImageManager::resize(
-            $filePath,
-            sprintf('%s-%s%s', rtrim($filePath, '.' . $fileExtension), stripslashes($imageType->name), $destinationExtension),
-            $width,
-            $height,
-            trim(mime_content_type($filePath), 'image/')
-        );
+        if (!$this->featureFlagRepository->isEnabled(FeatureFlagSettings::FEATURE_FLAG_MULTIPLE_IMAGE_FORMAT)) {
+            return ImageManager::resize(
+                $filePath,
+                sprintf('%s-%s%s', rtrim($filePath, '.' . $fileExtension), stripslashes($imageType->name), $destinationExtension),
+                $width,
+                $height,
+                trim(mime_content_type($filePath), 'image/'));
+        }
+
+        $result = true;
+
+        foreach ($this->imageFormatConfiguration->getGenerationFormats() as $imageFormat) {
+            if (!ImageManager::resize(
+                $filePath,
+                sprintf('%s-%s.%s', rtrim($filePath, '.' . $fileExtension), stripslashes($imageType->name), $imageFormat),
+                $width,
+                $height,
+                trim(mime_content_type($filePath), 'image/')
+            )) {
+                $result = false;
```

Comment:
if `result=false`, we can exit the foreach?

Reply:
I'm keeping the current pattern:

If an image can't be generated, result is false, an error message will be displayed, but we keep generating following image (different image sizes and formats)